### PR TITLE
Manager: Add migration note about react 18

### PR DESCRIPTION
**Pull request by ndelangen**
## What I did

I added a migration note informing addon creators about the move to react 18, and the API changes because of it.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -339,6 +340,26 @@
 
 ## From version 7.x to 8.0.0
 
+### Manager addons are now rendered with react 18
+
+The UI added to the manager via addon, are now rendered with react 18.
```
Comment (jonniebigodes) [inline MIGRATION.md:345]: @ndelangen when you have a moment can you adjust this sentence to be: 
"The UI added to the manager via addons is now rendered with React 18."

Also, can you capitalize react in the heading and adjust the link above accordingly?

**Review by jonniebigodes — APPROVED** [review]
@ndelangen, thank you for taking the time to put together this pull request and add this item to the migration. Appreciate it 🙏 ! I'll check the remainder of the documentation and adjust accordingly in a follow-up pull request. I left a small item for you to look into when you're able. To unblock this, I'm going to approve it so that you can merge it once the feedback is addressed.

File: MIGRATION.md (hunk 1 above)
Comment (ndelangen) [inline MIGRATION.md:345]:
```suggestion
The UI added to the manager via addons is now rendered with React 18.
```

File: MIGRATION.md [hunk 2]
```diff
@@ -339,6 +340,26 @@
 
 ## From version 7.x to 8.0.0
 
+### Manager addons are now rendered with react 18
```
Comment (ndelangen) [inline MIGRATION.md:343]:
```suggestion
### Manager addons are now rendered with React 18
```

File: MIGRATION.md [hunk 3]
```diff
@@ -1,7 +1,8 @@
 <h1>Migration</h1>
 
 - [From version 7.x to 8.0.0](#from-version-7x-to-800)
-  - [Framework specific vite plugins have to be explicitly added](#framework-specific-vite-plugins-have-to-be-explicitly-added)  
+  - [Manager addons are now rendered with react 18](#manager-addons-are-now-rendered-with-react-18)
```
Comment (ndelangen) [inline MIGRATION.md:4]:
```suggestion
  - [Manager addons are now rendered with React 18](#manager-addons-are-now-rendered-with-react-18)
```

**Pull request merged**